src/search/landmarks/exploration.cc

```diff
@@ -147,124 +122,91 @@ void Exploration::build_unary_operators(const OperatorProxy &op) {
-    int unsolved_goals = termination_propositions.size();
+void Exploration::relaxed_exploration() {
     while (!prop_queue.empty()) {
         pair<int, ExProposition *> top_pair = prop_queue.pop();
         int distance = top_pair.first;
         ExProposition *prop = top_pair.second;
 
-        int prop_cost;
-        if (use_h_max)
-            prop_cost = prop->h_max_cost;
-        else
-            prop_cost = prop->h_add_cost;
+        int prop_cost = prop->h_max_cost;
         assert(prop_cost <= distance);
         if (prop_cost < distance)
             continue;
-        if (!level_out && prop->is_termination_condition && --unsolved_goals == 0)
-            return;
         const vector<ExUnaryOperator *> &triggered_operators = prop->precondition_of;
         for (size_t i = 0; i < triggered_operators.size(); ++i) {
             ExUnaryOperator *unary_op = triggered_operators[i];
-            if (unary_op->h_add_cost == -2) // operator is not applied
+            if (unary_op->excluded) // operator is not applied
                 continue;
             --unary_op->unsatisfied_preconditions;
-            increase_cost(unary_op->h_add_cost, prop_cost);
             unary_op->h_max_cost = max(prop_cost + unary_op->base_cost,
                                        unary_op->h_max_cost);
-            unary_op->depth = max(unary_op->depth, prop->depth);
             assert(unary_op->unsatisfied_preconditions >= 0);
             if (unary_op->unsatisfied_preconditions == 0) {
-                int depth = unary_op->is_induced_by_axiom(task_proxy)
-                    ? unary_op->depth : unary_op->depth + 1;
-                if (use_h_max)
-                    enqueue_if_necessary(unary_op->effect, unary_op->h_max_cost,
-                                         depth, unary_op, use_h_max);
-                else
-                    enqueue_if_necessary(unary_op->effect, unary_op->h_add_cost,
-                                         depth, unary_op, use_h_max);
+                enqueue_if_necessary(unary_op->effect, unary_op->h_max_cost,
+                                     unary_op);
             }
         }
     }
 }
 
-void Exploration::enqueue_if_necessary(ExProposition *prop, int cost, int depth,
-                                       ExUnaryOperator *op,
-                                       bool use_h_max) {
+void Exploration::enqueue_if_necessary(ExProposition *prop, int cost,
+                                       ExUnaryOperator *op) {
     assert(cost >= 0);
-    if (use_h_max && (prop->h_max_cost == -1 || prop->h_max_cost > cost)) {
+    if (prop->h_max_cost == -1 || prop->h_max_cost > cost) {
         prop->h_max_cost = cost;
-        prop->depth = depth;
-        prop->reached_by = op;
-        prop_queue.push(cost, prop);
-    } else if (!use_h_max && (prop->h_add_cost == -1 || prop->h_add_cost > cost)) {
-        prop->h_add_cost = cost;
-        prop->depth = depth;
         prop->reached_by = op;
```

Comment:
Do we still use `reached_by` for anything? (Perhaps this code is still used to generate preferred operators?)

Reply:
Apparently not, I removed it.